iec61850serversimulator/src/main/java/com/cgi/iec61850serversimulator/functionclass/EventDataListener.java

```diff
@@ -329,6 +331,16 @@ public EventDataListener(final Device device, final Scheduler scheduler, final D
                     }
                 }
             }
+            // Iterate through toUpdateSchedules to update the schedules
+            for (Entry<RelaySchedulePair, Schedule> entry : toUpdateSchedules.entrySet()) {
+                RelaySchedulePair key = entry.getKey();
+                int relayNr = key.getRelayNr();
+                Schedule value = entry.getValue();
+                int scheduleNr = value.getIndexNumber();
+
+                this.databaseUtils.updateDatabaseSchedule(this.device.getRelay(relayNr).getSchedule(scheduleNr));
```

Comment:
This can be simplified. The entries in your Map are exactly the Schedule objects you need to update in the database.

Reply:
Done